File: src/main/java/com/d20charactersheet/framework/dac/dao/sql/SqlItemDao.java
package com.d20charactersheet.framework.dac.dao.sql;

import java.util.List;

import com.d20charactersheet.framework.boc.model.Armor;
import com.d20charactersheet.framework.boc.model.Good;
import com.d20charactersheet.framework.boc.model.Weapon;
import com.d20charactersheet.framework.boc.model.WeaponFamily;
import com.d20charactersheet.framework.dac.dao.ItemDao;

/**
 * Provides access to SQLite database on Android device storing items.
 */
public class SqlItemDao implements ItemDao {

  private final WeaponHelper weaponHelper;
  private final ArmorHelper armorHelper;
  private final GoodHelper goodHelper;

  /**
   * Creates DAO to access SQLite database on Android device with given helper.
   *
   * @param db
   *     The database to access.
   */
  public SqlItemDao(final Database db) {
    final SqlItemDaoHelper helper = new SqlItemDaoHelper(db);

    weaponHelper = new WeaponHelper(db, helper);
    armorHelper = new ArmorHelper(db, helper);
    goodHelper = new GoodHelper(db, helper);

  }

    @Override
    public List<Weapon> getAllWeapons(final List<WeaponFamily> allWeaponFamilies) {
        return weaponHelper.getAllWeapons(allWeaponFamilies);
    }

    @Override
    public List<Armor> getAllArmor() {
        return armorHelper.getAllArmor();
    }

    @Override
    public List<Good> getAllGoods() {
        return goodHelper.getAllGoods();
    }

    @Override
    public Weapon createWeapon(final Weapon weapon) {
        return weaponHelper.createWeapon(weapon);
    }

    @Override
    public void updateWeapon(final Weapon weapon) {
        weaponHelper.updateWeapon(weapon);
    }

    @Override
    public void deleteWeapon(final Weapon weapon) {
        weaponHelper.deleteWeapon(weapon);
    }

    @Override
    public Armor createArmor(final Armor armor) {
        return armorHelper.createArmor(armor);
    }

    @Override
    public void updateArmor(final Armor armor) {
        armorHelper.updateArmor(armor);
    }

    @Override
    public void deleteArmor(final Armor armor) {
        armorHelper.deleteArmor(armor);
    }

    @Override
    public Good createGood(final Good good) {
        return goodHelper.createGood(good);
    }

    @Override
    public void updateGood(final Good good) {
        goodHelper.updateGood(good);

    }

    @Override
    public void deleteGood(final Good good) {
        goodHelper.deleteGood(good);
    }

    @Override
    public List<WeaponFamily> getAllWeaponFamilies() {
        return weaponHelper.getAllWeaponFamilies();
    }

}

File: src/main/kotlin/com/d20charactersheet/framework/dac/dao/sql/SqlItemDao.kt
package com.d20charactersheet.framework.dac.dao.sql

import com.d20charactersheet.framework.boc.model.Armor
import com.d20charactersheet.framework.boc.model.Good
import com.d20charactersheet.framework.boc.model.Weapon
import com.d20charactersheet.framework.boc.model.WeaponFamily
import com.d20charactersheet.framework.dac.dao.ItemDao

/**
 * Provides access to SQLite database on Android device storing items.
 * Creates DAO to access SQLite database on Android device with given helper.
 *
 * @param db
 * The database to access.
 */
class SqlItemDao(db: Database) : ItemDao {

    private val helper = SqlItemDaoHelper(db)
    private val weaponHelper = WeaponHelper(db, helper)
    private val armorHelper = ArmorHelper(db, helper)
    private val goodHelper = GoodHelper(db, helper)

    override fun getAllWeapons(allWeaponFamilies: List<WeaponFamily>): List<Weapon> {
        return weaponHelper.getAllWeapons(allWeaponFamilies)
    }

    override fun getAllArmor(): List<Armor> {
        return armorHelper.allArmor
    }

    override fun getAllGoods(): List<Good> {
        return goodHelper.allGoods
    }

    override fun createWeapon(weapon: Weapon): Weapon {
        return weaponHelper.createWeapon(weapon)
    }

    override fun updateWeapon(weapon: Weapon) {
        weaponHelper.updateWeapon(weapon)
    }

    override fun deleteWeapon(weapon: Weapon) {
        weaponHelper.deleteWeapon(weapon)
    }

    override fun createArmor(armor: Armor): Armor {
        return armorHelper.createArmor(armor)
    }

    override fun updateArmor(armor: Armor) {
        armorHelper.updateArmor(armor)
    }

    override fun deleteArmor(armor: Armor) {
        armorHelper.deleteArmor(armor)
    }

    override fun createGood(good: Good): Good {
        return goodHelper.createGood(good)
    }

    override fun updateGood(good: Good) {
        goodHelper.updateGood(good)
    }

    override fun deleteGood(good: Good) {
        goodHelper.deleteGood(good)
    }

    override fun getAllWeaponFamilies(): List<WeaponFamily> {
        return weaponHelper.allWeaponFamilies
    }


}